chia/util/db_wrapper.py

```diff
@@ -70,39 +70,52 @@ def _next_savepoint(self) -> str:
         self._savepoint_name += 1
         return name
 
+    @contextlib.asynccontextmanager
+    async def _savepoint(self, connection) -> AsyncIterator[None]:
+        name = self._next_savepoint()
+        await connection.execute(f"SAVEPOINT {name}")
+        try:
+            # TODO: maybe yield out something to make it possible to cancel the
+            #       savepoint other than raising an exception?
```

Comment:
in my experience, aborting without collaborating with the holder of the resource is really questionable and unreliable (mostly really complex). If we cancel a task, and get a `TaskCancelled` exception, I would expect it all to just work out anyway.

That's a long way of saying; "I don't think we need this comment here".

Reply:
I didn't mean to replace the existing behavior, rather that augmenting it might be useful.  basically something to allow:

```python
async with wrapper.write_db() as connection:
    connection.execute(something)
    if condition():
        connection.undo_my_changes()
```

Instead of forcing raising of an exception through the context manager and subsequent handling of that except just to achieve an undo:

```python
try:
    async with wrapper.write_db() as connection:
        connection.execute(something)
        if condition():
            raise CustomErrorJustToCancelTransaction()
finally CustomErrorJustToCancelTransaction:
    pass
```

The comment isn't a big deal I suppose.  It seems like there may be cases where it would be a nice addition.  We'll see if they come up or not.  I guess maybe it would just be a rollback to the relevant savepoint.